src/charts/sparkline/sparkline-chart.directive.js

```diff
@@ -0,0 +1,327 @@
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <th colspan="2">' + d[0].x.toLocaleDateString() + '</th>' +
+                    '      </tr>' +
+                    '      <tr>' +
+                    '        <td class="name">' + dayUsage + '%:' + '</td>' +
+                    '        <td class="value text-nowrap">' + d[0].value + ' ' + scope.config.totalUnits + ' ' + d[0].name + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else if (scope.config.tooltipType === 'valuePerDay') {
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <td class="value">' +  d[0].x.toLocaleDateString() + '</td>' +
+                    '        <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else if (scope.config.tooltipType === 'percentage') {
+                  var percentUsed = Math.round(d[0].value / scope.chartData.total * 100.0);
+                  tip = '<div id="sparkline-tooltip" class="module-triangle-bottom">' +
+                    '  <table class="c3-tooltip">' +
+                    '    <tbody>' +
+                    '      <tr>' +
+                    '        <td class="name">' + percentUsed + '%' + '</td>' +
+                    '      </tr>' +
+                    '    </tbody>' +
+                    '  </table>' +
+                    '</div>';
+                }
+                else {
+                  tip = c3ChartDefaults.getDefaultSparklineTooltip().contents(d);
+                }
+                return tip;
+              },
+              position: function (data, width, height, element) {
+                try {
+                  var center = parseInt(element.getAttribute('x'));
+                  var top = parseInt(element.getAttribute('y'));
+                  var chartBox = document.querySelector('#' + scope.sparklineChartId).getBoundingClientRect();
+                  var graphOffsetX = document.querySelector('#' + scope.sparklineChartId + ' g.c3-axis-y').getBoundingClientRect().right;
+                  var x = Math.max(0, center + graphOffsetX - chartBox.left - Math.floor(width / 2));
+
+                  return {
+                    top: top - height,
+                    left: Math.min(x, chartBox.width - width)
+                  };
+                }
+                catch (TypeError) {
+                }
```

Comment:
not returning a value puts the tooltip at 0,0 which is probably what I would return anyhow. I could add that if you feel its more robust/readable